### ISSUE TITLE
Style Guide - Patterns

### ISSUE THREAD

**Issue by mheppler** [IQSS/dataverse#3755]
Second installment of the Style Guide, after Foundations #3635, the Patterns page will show the various types of UI elements (e.g. buttons, tables, forms) and how they are used across the application.

**Comment by TaniaSchlatter** [comment]
Suggest breaking the second installment of the Style Guide down:
1. Outlining the major page templates and pages (à la http://bradfrost.com/blog/post/atomic-web-design/): file page, dataset page, etc. Maybe this page is called "Templates & Pages" and is different from the Patterns page)
2. Patterns: Buttons & System messages
3. Patterns: Citations, Tables & Forms
(4. There may be more – not clear yet)

**Comment by djbrooke** [comment]
In Sprint Planning 4/6, we said we'd take on:

- Patterns: Buttons & System messages
- Patterns: Citations, Tables & Forms

We'll not take on:

- Outlining the major page templates and pages

I'll create another issue for the page templates and link it from here.

**Comment by pdurbin** [comment]
I just set up http://guides.dataverse.org/en/3755-style-guide-patterns/style/patterns.html to build from Jenkins. Anyone with access can click "Build Now" here to build the latest: https://build.hmdc.harvard.edu:8443/job/guides.dataverse.org-3755-style-guide-patterns/

**Comment by pdurbin** [comment]
I just build the docs locally as of bd2cd14 and they are really coming along. Good work, @mheppler . I just tweaked the Jenkins job above to poll for changes hourly so nobody needs to bother clicking "Build Now". As @mheppler mentioned at standup this morning, there are still some TO-DO's in the text itself that need to be addressed before we can move this issue to code review.

**Comment by pdurbin** [comment]
@mheppler @TaniaSchlatter @dlmurphy @djbrooke and I just agreed that "Link" should look like "Edit" and "Upload Files", buttons that only users with "write" access can click. I left a comment at https://github.com/IQSS/dataverse/commit/e975c12453b79dd02d9bb4b0c1feb04e8b44f3d5#commitcomment-21916346

**Comment by dlmurphy** [comment]
The Patterns page passes my review, with some small edits. Looking good!

**Comment by mheppler** [comment]
@kcondon In addition to the new Patterns pg, the Foundations pg was updated to include a download version of the newly registered Dataverse Project logo.
